components/transactions/TransactionRefundButton.js

```diff
@@ -21,65 +22,88 @@ export const refundTransactionMutation = gqlV2/* GraphQL */ `
 `;
 
 const TransactionRefundButton = props => {
-  const [refundTransaction, { loading, error }] = useMutation(refundTransactionMutation, {
+  const [refundTransaction, { error: mutationError }] = useMutation(refundTransactionMutation, {
     context: API_V2_CONTEXT,
   });
   const transactionsDataContext = React.useContext(GraphQLContext);
   const [isEnabled, setEnabled] = React.useState(false);
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    setError(mutationError);
+  }, [mutationError]);
 
   const handleRefundTransaction = async () => {
     await refundTransaction({ variables: { transaction: { id: props.id } } });
     setEnabled(false);
     transactionsDataContext?.refetch?.();
   };
 
+  const closeModal = () => {
+    setEnabled(false);
+    setError(null);
+  };
+
+  const rejectButton = () => {
+    return (
+      <StyledButton
+        buttonSize="small"
+        minWidth={140}
+        background="transparent"
+        textTransform="capitalize"
+        onClick={() => setEnabled(true)}
+        disabled={props.disabled}
+      >
+        <FormattedMessage id="transaction.refund.btn" defaultMessage="refund" />
+      </StyledButton>
+    );
+  };
```

Comment:
No need to write that as a function, you can simplify as:

```suggestion
  const rejectButton = (
      <StyledButton
        buttonSize="small"
        minWidth={140}
        background="transparent"
        textTransform="capitalize"
        onClick={() => setEnabled(true)}
        disabled={props.disabled}
      >
        <FormattedMessage id="transaction.refund.btn" defaultMessage="refund" />
      </StyledButton>
    );
```

And render it as:

```jsx
{rejectButton}
```